test: schedule_api : test some kernel api

yingmingx opened the pull request:
1.In order to improve function coverage, add testcase
for kernel api api z_pend_thread() z_unpend_thread()
z_unpend_all() z_pend_curr_irqlock()
z_move_thread_to_end_of_prio_q() z_remove_thread_from_ready_q()

2.Add error test of api in sched.c to improve
branch and line coverage.

Signed-off-by: Ying ming <mingx.ying@intel.com>

LeiW000 commented on tests/kernel/sched/schedule_api/src/test_sched_pending.c:
```diff
@@ -0,0 +1,189 @@
+
+static void thread_handler3(void *p1, void *p2, void *p3)
+{
+	num++;
```
I'm wondering if we should add a kind of lock here?

yingmingx replied:
This is really a problem that needs attention. I have changed code and used atomic variable here.

enjiamai commented on tests/kernel/sched/schedule_api/src/test_sched_pending.c:
```diff
@@ -0,0 +1,189 @@
+#ifdef CONFIG_ASSERT_NO_FILE_INFO
+void assert_post_action(void)
+#else
+void assert_post_action(const char *file, unsigned int line)
```
The common fatal and assert handler has been merged, could you please leverage them instead? thanks!  
please refer to https://github.com/zephyrproject-rtos/zephyr/pull/30150

enjiamai reviewed (changes requested):
The common fatal and assert handler has been merged, could you please leverage them instead? thanks!  
please refer to https://github.com/zephyrproject-rtos/zephyr/pull/30150

yingmingx commented:
> The common fatal and assert handler has been merged, could you please leverage them instead? thanks!
> please refer to #30150
Hi, enjia. All the assert related code have been removed and fatal framework have been used. Please help to give another look, thanks.

enjiamai commented on tests/kernel/sched/schedule_api/src/test_sched_pending.c:
```diff
@@ -0,0 +1,117 @@
+ * z_unpend_all() z_pend_curr_irqlock()
+ *
+ * @ingroup kernel_sched_tests
+ *
```
I know you have already commented the code. My suggestion is to add @details to descript the high level testing purpose of this case, thanks!

enjiamai commented on tests/kernel/sched/schedule_api/src/test_sched_pending.c:
```diff
@@ -0,0 +1,117 @@
+/**
+ * @brief Test kernel api z_move_thread_to_end_of_prio_q()
+ * z_remove_thread_from_ready_q()
+ *
```
Same as line 56.

yingmingx replied:
Hi ,enjia. Details tag has been added and all the build and run error solved.

enjiamai reviewed (changes requested):
could you please add @details for your test cases, thanks!

LeiW000 commented on tests/kernel/sched/schedule_api/src/test_sched_api_error.c:
```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
```
Let change it with "2021".

Commit:
test: schedule_api : test some kernel api

1.In order to improve function coverage, add testcase
for kernel api api z_pend_thread() z_unpend_thread()
z_unpend_all() z_pend_curr_irqlock()
z_move_thread_to_end_of_prio_q() z_remove_thread_from_ready_q()

2.Add error test of api in sched.c to improve
branch and line coverage.

Signed-off-by: Ying ming <mingx.ying@intel.com>

yingmingx commented:
Close this PR and replace it with #32601
